contrib/pyln-testing/pyln/testing/db.py

```diff
@@ -117,28 +116,24 @@ def __init__(self, directory):
         print("Starting PostgresDbProvider")
 
     def locate_path(self):
-        prefix = '/usr/lib/postgresql/*'
-        matches = glob(prefix)
-
-        candidates = {}
-        for m in matches:
-            g = re.search(r'([0-9]+[\.0-9]*)', m)
-            if not g:
-                continue
-            candidates[float(g.group(1))] = m
-
-        if len(candidates) == 0:
-            raise ValueError("Could not find `postgres` and `initdb` binaries in {}. Is postgresql installed?".format(prefix))
-
-        # Now iterate in reverse order through matches
-        for k, v in sorted(candidates.items())[::-1]:
-            initdb = os.path.join(v, 'bin', 'initdb')
-            postgres = os.path.join(v, 'bin', 'postgres')
-            if os.path.isfile(initdb) and os.path.isfile(postgres):
-                logging.info("Found `postgres` and `initdb` in {}".format(os.path.join(v, 'bin')))
+        # Use `pg_config` to determine correct PostgreSQL installation
+        pg_config = shutil.which('pg_config')
+        if not pg_config:
+            raise ValueError("Could not find `pg_config` to determine PostgreSQL binaries. Is PostgreSQL installed?")
+
+        proc = subprocess.Popen([pg_config, '--bindir'], stdout=subprocess.PIPE)
+        bindir = io.TextIOWrapper(proc.stdout).readline().rstrip()
```

Comment:
The easier way to get the output of a subprocess is to use `check_output`:
```suggestion
        bindir = subprocess.check_output([pg_config, '--bindir'], stdout=subprocess.PIPE).decode('ASCII').rstrip()
```

Reply:
Thx. I modified it slightly by skipping the `stdout=subprocess.PIPE` as it works without it for `check_output`. I think this is only required for error pipes etc. Also I left the `decode()` without argument, the default is UTF8 for that. Or do you think we need to decode it as ASCII for a specific reason? Maybe some very strange distribution is using path names that contain UTF8 chars for `pg_config` dirs...